### PR TITLE
SNI on backends

### PR DESCRIPTION
**Pull request by timricese**
Currently ORM does not work with backends which require SNI headers to be sent,
such as cloudflare, istio-gateway.

This fixes that.

### CODE REVIEW

**Comment by timricese** [comment]
@svtfrida @splushii Attention please!

**Comment by nsg** [comment]
This partially fixes some of the problems described in #21

File: orm/schemas/common-1.json [hunk 1]
```diff
@@ -201,6 +201,10 @@
                   "max_queued_connections": {
                     "type": "integer",
                     "minimum": 0
+                  },
+                  "sni": {
+                    "type": "string",
+		    "format": "hostname"
```
Comment (splushii) [inline orm/schemas/common-1.json:207]: Replace tabs with spaces

File: test/testdata/parser/rule_sni.yml [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+    domains:
+      - www.example.com
+    matches:
+      - any:
+          - paths:
+              equals:
+                - '/'
+    actions:
+      backend:
+        servers:
```
Comment (splushii) [inline test/testdata/parser/rule_sni.yml:16]: Trailing whitespace

File: build_requirements.txt [hunk 3]
```diff
@@ -4,16 +4,16 @@
...
-pip-tools==3.6.1
-pylint==2.3.1
-pyyaml==5.1               # via yamllint
-six==1.12.0               # via astroid, pip-tools
-typed-ast==1.3.5          # via astroid
-wrapt==1.11.1             # via astroid
-yamllint==1.15.0
+pathspec==0.8.0           # via yamllint
+pip-tools==3.9.0
+pylint==2.5.3
```
Comment (splushii) [inline build_requirements.txt:14]: New pylint version gives new lint errors (`make release-test`) which are good to fix.

File: test/test_parser.py [hunk 4]
```diff
@@ -44,6 +46,13 @@ def test_parse_yaml_file(self):
         for yml_doc in yml_docs:
             self.assertIsInstance(yml_doc, dict)
 
+    def test_sni(self):
+        yml_docs = parser.parse_yaml_file(self.valid_rule_file_sni)
+        for yml_doc in yml_docs:
+            exp_parsed_doc = {'backend': {'servers': [{'server': 'some-origin.example.com', 'sni': 'some-sni.example.com'}, {'server': 'some-potato-origin.example.com'}]}}
```
Comment (splushii) [inline test/test_parser.py:52]: Split to multiple lines for readability

File: build_requirements.txt (hunk 3 above)
Comment (timricese) [reply build_requirements.txt]: After fixing the first few pylint errors (basically using sys.exit() as in https://github.com/SVT/orm/pull/47/commits/d69d09ce2366176b71a53b86dad2fb384791d5a5) pylint complained about all the module imports from the test scripts

```
lint tests
************* Module test_rendervarnish
test/test_rendervarnish.py:5:0: E0401: Unable to import 'orm.render' (import-error)
test/test_rendervarnish.py:6:0: E0401: Unable to import 'orm.rendervarnish' (import-error)
test/test_rendervarnish.py:7:0: E0401: Unable to import 'orm.rendervarnish' (import-error)
************* Module test_validator
test/test_validator.py:5:0: E0401: Unable to import 'orm.validator' (import-error)
************* Module test_parser
test/test_parser.py:6:0: E0401: Unable to import 'orm.parser' (import-error)
************* Module test_renderhaproxy
test/test_renderhaproxy.py:3:0: E0401: Unable to import 'orm.render' (import-error)
test/test_renderhaproxy.py:4:0: E0401: Unable to import 'orm.renderhaproxy' (import-error)
test/test_renderhaproxy.py:5:0: E0401: Unable to import 'orm.renderhaproxy' (import-error)
```
As described in https://github.com/SVT/orm/pull/47/commits/28c4d5c256b40f98772f0217bc160ee1bfc7f62c this is due to changes with how pylint imports modules. I dont see any trivial way to fix this which wouldnt be potentially overstepping the scope of this pull request, so maybe downgrade pylint?

Comment (splushii) [reply build_requirements.txt]: Sounds good

**Pull request merged**